nms/app/views/events/EventsTable.js

```diff
@@ -262,11 +262,13 @@ export default function EventsTable(props: EventTableProps) {
 
   useEffect(() => {
     if (props.inStartDate && props.inEndDate) {
+      props.inStartDate !== undefined && setStartDate(props.inStartDate);
+      props.inEndDate !== undefined && setEndDate(props.inEndDate);
       if (tableRef.current) {
         tableRef.current.onQueryChange();
       }
     }
-  }, [props.inStartDate, props.inEndDate]);
```

Comment:
As I mentioned above to be able to to use `setStartDate` and `setEndDate` as dependencies we need to make sure that they are stable. And to so we need to stabilise the callback for `useRefreshingDateRange` i.e. wrap it into useCallback.

```
const {startDate, endDate, setStartDate, setEndDate} = useRefreshingDateRange(
   isAutoRefreshing,
   30000,
   useCallback(() => {
     tableRef.current?.onQueryChange();
   }, []),
);
```

Reply:
I made the suggested adjustments...